### PR TITLE
Miscellaneous Rules fixes

### PR DESCRIPTION
**Pull request by hobinjk**
Handle times consistently in UTC and improve rule name edit-clicking

### CODE REVIEW

File: src/rules-engine/triggers/TimeTrigger.js [hunk 1]
```diff
@@ -36,9 +36,10 @@ class TimeTrigger extends Trigger {
     let hours = parseInt(parts[0], 10);
     let minutes = parseInt(parts[1], 10);
 
+    // Time is specified in UTC
     let nextTime = new Date();
-    nextTime.setHours(hours);
-    nextTime.setMinutes(minutes);
+    nextTime.setUTCHours(hours);
```
Comment (mrstegeman) [inline src/rules-engine/triggers/TimeTrigger.js]: You can actually pass both hours and minutes to `setUTCHours`

File: static/js/rules/TimeTriggerBlock.js [hunk 2]
```diff
@@ -17,7 +17,9 @@ function TimeTriggerBlock(ruleArea, rule, x, y) {
 
   this.timeInput = document.createElement('input');
   this.timeInput.type = 'time';
-  this.timeInput.value = new Date().getHours() + ':' + new Date().getMinutes();
+  let hours = TimeTriggerBlock.leftPad(new Date().getHours());
```
Comment (mrstegeman) [inline static/js/rules/TimeTriggerBlock.js]: I'd suggest only getting one `Date()`. There's a small chance that these get called right at an hour transition, which could lead to a wrong time display.

File: static/js/rules/TimeTriggerBlock.js [hunk 3]
```diff
@@ -62,3 +65,38 @@ TimeTriggerBlock.prototype.onUp = function(clientX, clientY) {
     this.remove();
   }
 };
+
+TimeTriggerBlock.leftPad = function(n) {
+  if (n < 10) {
```
Comment (mrstegeman) [inline static/js/rules/TimeTriggerBlock.js]: You could use `n.toString().padStart(2, '0')`

File: static/js/rules/TimeTriggerBlock.js [hunk 4]
```diff
@@ -62,3 +65,38 @@ TimeTriggerBlock.prototype.onUp = function(clientX, clientY) {
...
+
+/**
+ * Convert from a utc time string to one in the local timezone
+ * @param {String} utcTime - formatted HH:MM
+ * @return {String}
+ */
+TimeTriggerBlock.utcToLocal = function(utcTime) {
+  let timeParts = utcTime.split(':');
+  let date = new Date();
+  date.setUTCHours(parseInt(timeParts[0]));
```
Comment (mrstegeman) [inline static/js/rules/TimeTriggerBlock.js]: Ditto above... You can pass both hours and minutes to `setUTCHours`.

File: static/js/rules/TimeTriggerBlock.js [hunk 5]
```diff
@@ -62,3 +65,38 @@ TimeTriggerBlock.prototype.onUp = function(clientX, clientY) {
...
+
+/**
+ * Convert from a local time string to one in UTC
+ * @param {String} localTime - formatted HH:MM
+ * @return {String}
+ */
+TimeTriggerBlock.localToUTC = function(localTime) {
+  let timeParts = localTime.split(':');
+  let date = new Date();
+  date.setHours(parseInt(timeParts[0]));
```
Comment (mrstegeman) [inline static/js/rules/TimeTriggerBlock.js]: Ditto above... You can pass both hours and minutes to `setHours`.

**Comment by hobinjk** [comment]
Thanks for the review! Today I have learned many things.

**Comment by hobinjk** [comment]
@mrstegeman Just noticed I missed one of the instances of setUTCHours. This should be good-to-go now

**Pull request merged**